react/components/Image.js

```diff
@@ -1,15 +1,39 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
+import { withRuntimeContext } from 'render'
+
+const MOBILE_SIZE = 200
+const DESKTOP_SIZE = 500
 
 /** Image component with 1:1 aspect ratio */
-export default function Image({ src, className }) {
-  return (
-    <div
-      className={classNames(className, 'vtex-aspect-ratio w-100')}
-      style={{ backgroundImage: `url(${src})` }}
-    />
-  )
+class Image extends Component {
+
+  resizeUrl = src => {
+    //Check of src is already in a size defined by the client
+    const resized = /[1-9]+-(:?([1-9]+)|(auto))/
+    if (resized.exec(src)) {
+      return src
+    }
+
+    const { runtime: { hints: { mobile } } } = this.props
+
+    const imageId = src.slice(src.indexOf('/ids/')).split('/')[2]
```

Comment:
Duplicated code from https://github.com/vtex-apps/product-details/blob/master/react/utils/generateUrl.js#L22.

@brunojdo do you agree we should have a HOC in `store-components` that executes this behavior?

Reply:
Add TODO, Technical Debt